src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -84,6 +87,8 @@ private void updateExercises() {
       List<ExerciseGroup> exerciseGroups = dataSource.getExerciseGroups(course, points, auth);
       inGrading.forEach((id, exercise) -> setInGrading(exerciseGroups, id));
       exercisesViewModel.set(new ExercisesTreeViewModel(exerciseGroups, exerciseFilterOptions));
+      exercisesViewModel.get().setAuthenticated(auth != null);
```

Comment:
`auth` is never `null` at this point, because the earlier `if (course == null || auth == null)` ends with a return.